src/libostree/ostree-repo-checkout.c

```diff
@@ -890,6 +890,39 @@ ostree_repo_checkout_tree_at (OstreeRepo                         *self,
   return ret;
 }
 
+/**
+ * ostree_repo_checkout_tree_at_introspectable:
+ * @self: Repo
+ * @options: (allow-none): Options
+ * @destination_dfd: Directory FD for destination
+ * @destination_path: Directory for destination
+ * @commit: Checksum for commit
+ * @cancellable: Cancellable
+ * @error: Error
+ *
+ * Introspectable version of ostree_repo_checkout_tree_at.
```

Comment:
Eeek.  That's a lot of new duplicate API to maintain.  I think if we're going to go this route we need to basically deprecate the earlier versions.  One way to do this would be to create e.g. `ostree-repo-deprecated.h` (still `#include'd` by `ostree-repo.h`).

Then we need to bikeshed on the name a bit...if this is the new canonical API then let's call it something other than "introspectable".  I'm blanking a bit...maybe just go all Microsoft and call it `ostree_repo_checkout_tree2`?  Or perhaps just `ostree_repo_checkout()`?